static/js/util.js

```diff
@@ -230,6 +230,20 @@ exports.is_all_or_everyone_mentioned = function (message_content) {
     return all_everyone_re.test(message_content);
 };
 
+exports.move_array_elements_to_front = function util_move_array_elements_to_front(array, selected) {
+    var selected_hash = {};
+    for (var i = 0; i < selected.length; i+=1) {
+        selected_hash[selected[i]] = true;
+    }
+    var selected_elements = array.filter(function (element) {
+        return selected_hash[element];
+    });
+    var unselected_elements = array.filter(function (element) {
+        return !selected_hash[element];
+    });
+    return selected_elements.concat(unselected_elements);
```

Comment:
Better to do it in two passes than adding elements to the beginning, as adding elements to the beginning of an array has to shift all other elements one forward (though if there were a list type in JavaScript, all would be simpler).